muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/ReferenceMatcher.java

```diff
@@ -283,7 +286,9 @@ private static MethodDescription.InDefinedShape findMethod(
     for (MethodDescription.InDefinedShape methodDescription :
         typeOnClasspath.getDeclaredMethods()) {
       if (methodDescription.getInternalName().equals(methodRef.getName())
-          && methodDescription.getDescriptor().equals(methodRef.getDescriptor())) {
+          && matchesArgumentTypes(methodRef.getArgumentTypes(), methodDescription.getParameters())
+          && matchesReturnType(
```

Comment:
I don't understand how this would work. JVM invokes method only when the descriptor matches exactly. If we allow covariant return type here this could result in `NoSuchMethodError` during runtime unless the descriptor used for invocation is somehow also rewritten.

Reply:
Yeah, you're right, I didn't think of that -- loosening up the muzzle check a little is definitely not enough. Back to the drawing board I guess